core/modules/cart/store/actions.ts

```diff
@@ -252,10 +251,8 @@ const actions: ActionTree<CartState, RootState> = {
         Logger.info('Cart token received from cache.', 'cache', token)()
         Logger.info('Syncing cart with the server.', 'cart')()
         dispatch('sync', { forceClientState, dryRun: !config.cart.serverMergeByDefault })
-      } else {
-        Logger.info('Creating server cart token', 'cart')()
-        await dispatch('connect', { guestCart: false })
       }
```

Comment:
Shouldn’t we have else condition in here OR remove “sync” call above? Isn’t this form causing double sync? It probably is because create calls connect which calls sync for the second time. Isn’t it?

Reply:
we are checking in `create` if token already exist https://github.com/DivanteLtd/vue-storefront/pull/3994/files#diff-2101c92c40c144e3e56fb400ac211e3dR505 So if in above if statement we add token, then in `create` it is not called anymore.

Reply:
Ok 👌